kip-0017.md

```diff
@@ -60,53 +111,304 @@ kadena:mainnet01:22ddc64851718e9d41d98b0f33d5e328ae5bbbbd97aed9885adac0f2d070ff9
+
+> NB "accounts" is semantically specific to WalletConnect. This does not
+> represent an account in Kadena. The information returned is the `publickey` as
+> shown in [WalletConnectAccountAddress](#walletconnectaccountaddress)
+
+### Example Pairing Proposal and Response
+
+```json
+// Proposal namespace request
+{
+  "requiredNamespaces": {
+    "kadena": {
+      "chains": ["kadena:mainnet01", "kadena:testnet04", "kadena:development"],
+      "methods": ["kadena_get_accounts_v1", "kadena_sign_v1", "kadena_quicksign_v1"],
+      "events": []
+    }
+  }
+}
+```
 
 ```json
-# Proposal namespace request
+// Session namespace response
 {
-  "kadena": {
-    "chains": ["kadena:mainnet01", "kadena:testnet04", "kadena:development"],
-    "methods": ["kadena_quicksign_v1"],
-    "events": []
+  "namespaces": {
+    "kadena": {
+      "accounts": [
+        "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+        "kadena:testnet04:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+        "kadena:testnet04:22ddc64851718e9d41d98b0f33d5e328ae5bbbbd97aed9885adac0f2d070ff9c"
+      ],
+      "methods": ["kadena_get_accounts_v1", "kadena_sign_v1", "kadena_quicksign_v1"],
+      "events": []
+    }
   }
 }
 ```
 
+### Accounts
+
+The accounts property is an Array of
+*[`WalletConnectAccountString`](#walletconnectaccountstring)*s
+
+### WalletConnectAccountString
+
+This is the description of an account provided by the wallet
+
+A **WalletConnectAccountString** is build from
+[`WalletConnectChainID`](#walletconnectchainid)`:`[`WalletConnectAccountAddress`](#walletconnectaccountaddress)
+
+#### WalletConnectChainID
+
+This is the description of the chain. It's build from
+[`WalletConnectNamespace`](#walletconnectnamespace)`:`[`WalletConnectReference`](#walletconnectreference)
+
+#### WalletConnectNamespace
+
+The namespace for Kadena is `kadena`
+
+#### WalletConnectReference
+
+Is one of:
+
+- `mainnet01`
+- `testnet04`
+- `development`
+
+#### WalletConnectAccountAddress
+
+This is the Kadena `publickey` that the wallet has access to. In some cases the
+account belonging associated with the publickey in a fungible contract e.g.
+`coin` can be derived from the publickey, in other cases to get the account to
+use with this publickey the [`kadena_get_accounts_v1`](#kadena_get_accounts_v1)
+method can be called.
+
+## WalletConnect Methods
+
+The methods available are:
+
+- [`kadena_get_accounts_v1`](#kadena_get_accounts_v1)
+- [`kadena_quicksign_v1`](#kadena_quicksign_v1)
+- [`kadena_sign_v1`](#kadena_sign_v1)
+
+### kadena_get_accounts_v1
+
+From the [Paring response](#pairing-with-walletconnect) an element from the
+`accounts` property can be used to retrieve the available KadenaAccounts
+
+#### Example kadena_get_accounts_v1 Request and Response
+
+```jsonc
+// kadena_get_accounts_v1 request
+{
+  "accounts": [
+    "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89"
+  ]
```

Comment:
I think these properties are missing from the request example:
```suggestion
       "accounts":[{
        "account": "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
        "contracts": ["coin"]
      }]
```

Reply:
Yes I've added them.